### DISCUSSION TITLE
Problem with ESP32 and GC9A01 Touch screen

### DISCUSSION THREAD

**Discussion by AdmiralAki99** [Bodmer/TFT_eSPI#2568] [Q&A - General] [answered]
I was successful in connecting the display to the ESP32 but the program crashes when the screen is touched. I using the touch_cs pin as -1 but it leads to GPIO output gpio_num error. What can I do about this? I am not very familiar with using touch screens. Thanks

**Comment by Bodmer** [comment]
Which touch controller does the display use?
The library only supports XPT2046 type.
If the controller is a different type then an extra library will be needed, in that case do not define TOUCH_CS.

**Reply by AdmiralAki99** [reply]
The touch driver based on the specifications on the website is CST816S. Do I have to download an external library? Thanks for the reply

**Reply by Bodmer** [reply]
Yes

**Reply by AdmiralAki99** [reply]
Thank you very much! , yes you were correct I needed a separate library for it to work. When I try to initialize it and tft_espi together the screen does not light up but the touch is still working. Is that something to do with the pins that I might have selected? Sorry for the question, I am not able to figure it out

**Reply by Bodmer** [reply]
Possibly but you don't say how it is wired up or what your software environment is. I am not clairvoyant!

**Reply by AdmiralAki99** [reply]
The pin config of the screen and the board is:


ESP32 <--> TFT Touch Screen (GC9A01)
VCC    <-->  VCC
GND    <-->  GND
MISO   <-->  GPIO 22
MOSI   <-->  GPIO 23
SCLK    <--> GPIO 18
LCD_CS <--> GPIO 15
LCD_DC<--> GPIO 2
LCD_RST <--> GPIO 4
LCD_BL <---> VCC (Not Controlling Backlight)
TP_SDA <--> GPIO 21
TP_SCL <--> GPIO 22
TP_INT <--> GPIO 3
TP_RST <-->GPIO 5

and the User_Setup.h for TFT_eSPI library looks like:

// For ESP32 Dev board (only tested with GC9A01 display)
// The hardware SPI can be mapped to any pins

#define TFT_MOSI 23 // In some display driver board, it might be written as "SDA" and so on.
#define TFT_SCLK 18
#define TFT_CS   15  // Chip select control pin
#define TFT_DC   2  // Data Command control pin
#define TFT_RST  4  // Reset pin (could connect to Arduino RESET pin)
#define TFT_BL   35  // LED back-light (Random Pin)

The screen shows the output (boing_ball example) when TP_INT is connected but the brightness is very low and I had to use a flashlight to check if it was working, but the touch panel was functioning. However, when the TP_INT was plugged into the positive line on the breadboard and not in the corresponding pinhole for GPIO 3 the brightness is correct and everything is visible perfectly but the touch panel does not work.

**Reply by Bodmer** [reply, marked as answer]
MISO and TP_SCL should not be connected to the same GPIO. That will prevent the touch screen working.

GPIO 35 is input only so cannot be used to control the backlight.

Connecting TP_INT  to a supply rail is not a good idea, it may damage the touch controller.

It appears the backlight is not being powered correctly. Look at the connections and check the display power lines have the correct voltage.
